### PR TITLE
feat: add processor to send notifications to ms teams

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

A new processor that can be used to send notifications to Microsoft Teams webhooks. The webhook must be either entity annotation or for broadcast in the configuration.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by drodil** [comment]
The absolute link utility is the same as here #24555 which needs refactoring once this is merged.

**Comment by drodil** [comment]
Not stale, @mareklibra chance to review?

**Review by mareklibra — APPROVED** [review]
I haven't run it but code-wise it lgtm.

**Comment by drodil** [comment]
Rebased

**Comment by drodil** [comment]
Not stale

**Comment by mareklibra** [comment]
@drodil Let's merge it when the Ci is green.

**Comment by drodil** [comment]
@mareklibra we are green!

**Review by mareklibra — APPROVED** [review]

File: plugins/notifications-backend-module-teams/src/module.ts
Comment (freben) [inline plugins/notifications-backend-module-teams/src/module.ts:41]: could you depend on [`catalogServiceRef`](https://github.com/backstage/backstage/blob/18374ff9c00dfb2e72182d20e161e1ffa3aed833/plugins/catalog-node/src/catalogService.ts#L28) instead?

File: plugins/notifications-backend-module-teams/src/processor/NotificationsTeamsProcessor.test.ts
Comment (freben) [inline plugins/notifications-backend-module-teams/src/processor/NotificationsTeamsProcessor.test.ts:86]: this triplet exists as [`setupRequestMockHandlers`](https://github.com/backstage/backstage/blob/18374ff9c00dfb2e72182d20e161e1ffa3aed833/packages/backend-test-utils/src/msw/setupRequestMockHandlers.ts#L22) in backend-test-utils

File: plugins/notifications-backend-module-teams/src/processor/NotificationsTeamsProcessor.ts
Comment (freben) [inline plugins/notifications-backend-module-teams/src/processor/NotificationsTeamsProcessor.ts:273]: [always use](https://github.com/backstage/backstage/blob/master/docs/architecture-decisions/adr013-use-node-fetch.md) the `node-fetch` version of fetch on the backend

**Comment by drodil** [comment]
Not stale, waiting for review/merge

**Comment by freben** [comment]
@drodil note review comments above - also unfortunate conflict

(i'm back now! 👋)

**Comment by drodil** [comment]
Hey @freben , thanks for the comments, I will check those! And welcome back 😎

**Comment by drodil** [comment]
Actually it does not make sense to fix and merge this PR as the webhooks are going to be deprecated by M$: https://devblogs.microsoft.com/microsoft365dev/retirement-of-office-365-connectors-within-microsoft-teams/

Have to figure out the new way of sending notifications to Teams.

**Pull request closed without merging**
